.circleci/config.yml

```diff
@@ -44,6 +44,75 @@ jobs:
           working_directory: ./web
           command: yarn test:unit
 
+  test_e2e:
+    docker:
+      - image: circleci/python:3.6-node-browsers
+      - image: viime/opencpu-base
+        command: |
+          /bin/sh -c "chmod a+w /usr/local/lib/R/site-library && service cron start && /usr/lib/rstudio-server/bin/rserver && apachectl -DFOREGROUND"
+    steps:
+      - checkout
+      - run:
+          name: Create a virtualenv for installation
+          command: |
+            python3 -m venv install_env
+            . install_env/bin/activate
+      - run:
+          name: Upgrade pip
+          command: pip install -U pip
+      - run:
+          name: Install python
+          command: |
+            pip install .
+            pip install sentry-sdk[flask]
+            pip install --upgrade sentry-sdk
+      - run:
+          working_directory: ./viime
+          name: Start the development server
+          command: flask run
+          background: true
+      # docker is broken and needs to be fixed but CI succeeded               systemctl --user enable docker
+      - run:
+          working_directory: ./devops
+          name: Run docker
+          command: |
+              docker ps
+              docker build -t viime .
+              docker run -it --rm -p 8004:8004 viime
+          background: true
+      - run:
+          working_directory: ./web
+          name: Install dependencies
+          command: yarn
+      - run:
+          working_directory: ./web
+          name: Serve the website
+          command: yarn serve
+          background: true
+      - restore_cache:
+          name: Restore Yarn Package Cache
+          keys:
+            - yarn-packages-{{ checksum "e2e-test/package.json" }}
```

Comment:
After thinking about this I think caching is definitely out of scope for this PR. The [docs](https://circleci.com/docs/2.0/caching/) should be read thoroughly before adding caching; this snippet copied from the `test_web` job will, I think, create cache collisions between the two jobs. It seems like if we're going to be caching the `e2e-test` build files we might as well cache everything. Additionally, the caches can be shared between the different build steps, i.e. the build from `test_web` can be cached and reloaded here in `test_e2e`.
Doing caching properly would take a substantial amount of work to tune up, so it should be done later if CI latency is ever an issue.